Library/Homebrew/cleanup.rb

```diff
@@ -39,6 +39,12 @@ def nested_cache?
       directory? && %w[go_cache glide_home java_cache npm_cache gclient_cache].include?(basename.to_s)
     end
 
+    def needs_chowning?
+      # Go makes its cache contents read-only to ensure cache integrity,
+      # which makes sense but is something we need to undo for cleanup.
+      directory? && %w[go_cache].include?(basename.to_s)
```

Comment:
This doesn't check if it needs chowning or not? Maybe worth checking if it's readable?

Reply:
Could just tweak the name; `is_go_cache_directory?` or similar I think signals what it does a bit more?

Reply:
Sure, if you're happy with that it works for me.

I generalised it in case we needed to expand it to other tools/formulae/etc for whatever reason in the future, but we can always handle that in the future rather than me trying to peer into a 🔮, heh.

Reply:
Aye, can refactor if and when we need it for more 👍